src/components/select/select.js

```diff
@@ -1,30 +1,65 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import PropTypes from "prop-types";
+import deprecated from "deprecated-prop-type";
+import warning from "warning";
 
 /**
  *
  * @param {string} label
  * @param {string} description
- * @param {[{ text: string, value: string, selected?:boolean }]}
- * options List of select options with key, value and selected properties
+ * @param {[{ text: string, value: string | number, disabled?: boolean }]} options
+ * List of select options with key, value and disabled properties
  * @param {object} selectProps Contains HTML input attributes:
  * type, value, name, id, etc. https://www.w3schools.com/tags/tag_select.asp
+ * @param { string | number } defaultValue
  */
-export const Select = ({ label, description, options, selectProps }) => {
+export const Select = ({
+  label,
+  description,
+  options,
+  selectProps,
+  defaultValue
+}) => {
+  const [currentValue, setCurrentValue] = useState();
+
+  useEffect(() => {
+    const selectedOptions = options.filter(opt => opt.selected);
+
+    if (selectedOptions.length) {
+      const [option] = selectedOptions;
+      setCurrentValue(option.value);
+
+      if (selectedOptions.length > 1) {
+        warning(false, "Only one 'selected' property of 'Select' can be true");
+      }
+    }
+
+    if (defaultValue) {
```

Comment:
Do we want the default value to always override the selected value if present ?

Reply:
I think that this makes more sense. Considering the `defaultValue` is the latest implementation, people might use `defaultValue` and still have `option.selected` in their data structure, therefore overriding it would do just fine while this usage would be the way to go in the future. Does this make sense? 🤔

Reply:
Ah, this initializes the component with the specification form the JSON, not with form data from users. 

Clarifies my both last comments, thanks 👍